src/macaron/slsa_analyzer/database_store.py

```diff
@@ -0,0 +1,105 @@
+    AnalysisTable,
+    CheckFactsTable,
+    CheckResultTable,
+    RepositoryAnalysis,
+    SLSARequirement,
+)
+from macaron.util import logger
+
+
+def store_analyze_context_to_db(
+    table_name: str, db_man: DatabaseManager, analysis: AnalysisTable, analyze_ctx: AnalyzeContext
+) -> dict:
+    """Store the content of an analyzed context into the database.
+
+    Parameters
+    ----------
+    table_name: str
+        The name of the main analysis table
+    db_man : DatabaseManager
+        The database manager object managing the session to which to add the results.
+    analysis: AnalysisTable
+        The analysis record which this result belongs to.
+    analyze_ctx : AnalyzeContext
+        The analyze context to store into the database.
+    """
+    logger.info(
+        "Inserting result of %s to %s",
+        analyze_ctx.repo_full_name,
+        defaults.get("database", "db_name", fallback="macaron.db"),
+    )
+
+    # Ensure result table is created
+    result_table = AnalyzeContext.get_analysis_result_table(table_name)
+    db_man.create_tables()
+
+    # Store old result format
+    repository_analysis = RepositoryAnalysis(repository_id=analyze_ctx.repository_table.id, analysis_id=analysis.id)
+    db_man.add_and_commit(repository_analysis)
+
+    # Store the context's slsa level
+    db_man.add_and_commit(analyze_ctx.get_slsa_level_table())
+
+    # Store check result table
+    for check in analyze_ctx.check_results.values():
+
+        check_table = CheckResultTable()
+        check_table.check_id = check["check_id"]
+        check_table.repository = analyze_ctx.repository_table.id
+        check_table.passed = check["result_type"] == CheckResultType.PASSED
+        check_table.skipped = check["result_type"] == CheckResultType.SKIPPED
+        db_man.add(check_table)
+
+        if "result_tables" in check:
+            for table in check["result_tables"]:
+                if isinstance(table, CheckFactsTable):
+                    table.repository = analyze_ctx.repository_table.id
+                    table.check_result = check_table.id
+                db_man.add_and_commit(table)
+
+    # Store SLSA Requirements
+    results = analyze_ctx.get_analysis_result_data()
+    for key, value in analyze_ctx.ctx_data.items():
+        if value.is_pass:
+            requirement = SLSARequirement(
+                repository=analyze_ctx.repository_table.id,
+                requirement=key.name,
+                requirement_name=value.name,
+                feedback=value.feedback,
+            )
+            db_man.add_and_commit(requirement)
+
+    db_man.insert(result_table, results)
+    return results
+
+
+def store_analysis_to_db(db_man: DatabaseManager, main_record: Record) -> AnalysisTable:
+    """Store the analysis to the database."""
+    db_man.create_tables()
+
+    analysis = AnalysisTable(
+        analysis_time=datetime.now().isoformat(sep="T", timespec="seconds"),
```

Comment:
Two issues with using `now` (in general):

1. It gives the local time of host machine and is not normalized to UTC.
2. It doesn't have a timezone attached.

```suggestion
        analysis_time=datetime.utcnow(tz=datetime.timezone.utc).isoformat(sep="T", timespec="seconds"),
```

Reply:
It looks like using `datetime.now(tz=timezone.utc)` is the recommended way to represent UTC while retaining the timezone information,https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow

Reply:
Makes sense, thanks.

Reply:
Fixed here https://github.com/oracle-samples/macaron/pull/46/commits/4e61112595339f7871796e707f604cc1ff1b5711